### PR TITLE
Refactor PendingTradesViewModel methods -> TradeUtil & OfferUtil

### PR DESCRIPTION
**Pull request by ghubstan**
PR https://github.com/bisq-network/bisq/pull/4697 should be reviewed & merged before this PR.

### CODE REVIEW

File: desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java [hunk 1]
```diff
@@ -199,55 +205,30 @@ private void onMessageStateChanged(MessageState messageState) {
...
-    }
-
-    private long getRemainingTradeDuration() {
-        return getMaxTradePeriodDate() != null ? getMaxTradePeriodDate().getTime() - new Date().getTime() : getMaxTradePeriod();
+        return item == null ? "" : tradeUtil.getMarketDescription(item.getTrade());
     }
 
     public String getRemainingTradeDurationAsWords() {
-        return FormattingUtils.formatDurationAsWords(Math.max(0, getRemainingTradeDuration()));
+        return tradeUtil.getRemainingTradeDurationAsWords(dataModel.getTrade());
```
Comment (sqrrm) [inline desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java:218]: I think a check for non null Trade would be good here.

File: desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java [hunk 2]
```diff
@@ -199,55 +205,30 @@ private void onMessageStateChanged(MessageState messageState) {
...
-        if (maxPeriod != 0) {
-            return 1 - (double) remaining / (double) maxPeriod;
-        } else
-            return 0;
+        return tradeUtil.getRemainingTradeDurationAsPercentage(dataModel.getTrade());
     }
 
     public String getDateForOpenDispute() {
-        return DisplayUtils.formatDateTime(new Date(new Date().getTime() + getRemainingTradeDuration()));
+        return DisplayUtils.formatDateTime(tradeUtil.getDateForOpenDispute(dataModel.getTrade()));
```
Comment (sqrrm) [inline desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java:226]: Also no check for null Trade

File: core/src/main/java/bisq/core/trade/TradeUtil.java [hunk 3]
```diff
@@ -108,4 +118,90 @@ public TradeUtil(BtcWalletService btcWalletService, KeyRing keyRing) {
...
+            return "";
+
+        checkNotNull(trade.getOffer());
+        checkNotNull(trade.getOffer().getCurrencyCode());
+        return getCurrencyPair(trade.getOffer().getCurrencyCode());
+    }
+
+    public String getPaymentMethodNameWithCountryCode(Trade trade) {
+        String paymentMethodDescription = "";
+        if (trade != null) {
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/trade/TradeUtil.java:167]: Prefer to return early, ie
```
if (trade == null)
    return;

Offer offer = checkNotNull(trade.getOffer());
checkNotNull(offer.getPaymentMethod());
return offer.getPaymentMethodNameWithCountryCode();
```

File: desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java (hunk 1 above)
Comment (ghubstan) [reply desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java:218]: Fixes for these problems will go into a new 'reviewer's changes' PR based on the last branch in the chain.
OK with you?

Comment (sqrrm) [reply desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java:218]: Acceptable

**Review by sqrrm — APPROVED** [review]
utACK

File: desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java (hunk 1 above)
Comment (ghubstan) [reply desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java:218]: Resolved in PR https://github.com/bisq-network/bisq/pull/4731, commit https://github.com/bisq-network/bisq/pull/4731/commits/0656c57d393e3313050a762effefa010347da970.

File: desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java (hunk 2 above)
Comment (ghubstan) [reply desktop/src/main/java/bisq/desktop/main/portfolio/pendingtrades/PendingTradesViewModel.java:226]: Resolved in PR https://github.com/bisq-network/bisq/pull/4731, commit https://github.com/bisq-network/bisq/pull/4731/commits/0656c57d393e3313050a762effefa010347da970.

File: core/src/main/java/bisq/core/trade/TradeUtil.java (hunk 3 above)
Comment (ghubstan) [reply core/src/main/java/bisq/core/trade/TradeUtil.java:167]: Resolved in PR https://github.com/bisq-network/bisq/pull/4731, commit https://github.com/bisq-network/bisq/pull/4731/commits/f764e9feb07f1cae45309b6271ef84862ee7d8b4.

**Pull request merged**